logstash-core/spec/support/shared_contexts.rb

```diff
@@ -56,7 +56,7 @@ def block_until(limit_seconds, &condition)
 
   before :all do
     clear_data_dir
-    settings = mock_settings("config.reload.automatic" => true)
+    settings = mock_settings("config.reload.automatic" => true, "queue.type" => "persisted")
```

Comment:
TODO: don't enable the PQ for all of the shared tests in order to validate behavior of the PQ in just one of those tests. I recently worked with this shared context so I can see if I can find a way to limit the scope in which we have to incur the cost of spooling up PQ's.

Reply:
PR here -> https://github.com/mashhurs/logstash/pull/4

Reply:
Great approach to apply specific setting through the context: 
```
include_context "api setup", {"queue.type" => "persisted"}
```